# HG changeset patch
# User Rik <rik@octave.org>
# Date 1513192677 28800
#      Wed Dec 13 11:17:57 2017 -0800
# Node ID 66977aecf3c00d1f92f3e85deeec2a6ff1c34701
# Parent  ca0ebd1d0da1014821f783cca417552bbc082f85
Fix error when setting "units" property on hggroup w/gnuplot (bug #39394).

* __gnuplot_draw_axes__.m: Only change obj "units" property to "data" if it is
a text object.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -508,17 +508,17 @@ function __gnuplot_draw_axes__ (h, plot_
     endif
     if (isfield (obj, "ydata"))
       obj.ydata = double (obj.ydata);
     endif
     if (isfield (obj, "zdata"))
       obj.zdata = double (obj.zdata);
     endif
 
-    if (isfield (obj, "units"))
+    if (strcmp (obj.type, "text"))
       units = obj.units;
       unwind_protect
         set (h_obj, "units", "data");
         obj = get (h_obj);
       unwind_protect_cleanup
         set (h_obj, "units", units);
       end_unwind_protect
     endif
